src/ChainSafe.Gaming.MetaMask.Unity/MetaMaskSigner.cs

```diff
@@ -0,0 +1,94 @@
+using System;
+using System.Threading.Tasks;
+using ChainSafe.Gaming.Evm.Signers;
+using ChainSafe.Gaming.Web3.Core;
+using ChainSafe.Gaming.Web3.Core.Evm;
+using ChainSafe.Gaming.Web3.Environment;
+using Nethereum.RPC.AccountSigning;
+using Nethereum.Unity.Rpc;
+using Newtonsoft.Json;
+
+namespace ChainSafe.Gaming.MetaMask.Unity
+{
+    /// <summary>
+    /// Implementation of <see cref="ISigner"/> for Metamask.
+    /// </summary>
+    public class MetaMaskSigner : ISigner, ILifecycleParticipant
+    {
+        private readonly IMetaMaskProvider metaMaskProvider;
+        private readonly ILogWriter logWriter;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MetaMaskSigner"/> class.
+        /// </summary>
+        /// <param name="metaMaskProvider">Metamask provider that connects to Metamask and makes JsonRPC requests.</param>
+        /// <param name="logWriter">Log Writer used for logging messages and errors.</param>
+        public MetaMaskSigner(IMetaMaskProvider metaMaskProvider, ILogWriter logWriter)
+        {
+            this.metaMaskProvider = metaMaskProvider;
+            this.logWriter = logWriter;
+        }
+
+        /// <summary>
+        /// Signer's public key/address.
+        /// </summary>
+        public string Address { get; private set; }
```

Comment:
We can keep this private as there is already a GetAddress() for ISigner.

Reply:
yep, will do

Reply:
done